### PR TITLE
Added configuration option to use compression from the READ API for Arrow

### PR DESCRIPTION
**Pull request by VinayLondhe14**
Fix for issue #481

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -582,6 +582,18 @@ The API Supports a number of options to configure the read
      </td>
      <td>Read</td>
    </tr>
+   <tr valign="top">
```
Comment (davidrabinowitz) [inline README.md:585]: Please split to another PR, otherwise the documentation will be available before the release

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java [hunk 2]
```diff
@@ -82,6 +87,12 @@
   public static final int MIN_BUFFERED_RESPONSES_PER_STREAM = 1;
   public static final int MIN_STREAMS_PER_PARTITION = 1;
   private static final int DEFAULT_BIGQUERY_CLIENT_RETRIES = 10;
+  private static final String ARROW_COMPRESSION_CODEC_OPTION = "arrowCompressionCodec";
+  private static final ImmutableList<String> PERMITTED_ARROW_COMPRESSION_CODECS =
```
Comment (davidrabinowitz) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java:91]: Effectively duplicating `ArrowSerializationOptions.CompressionCodec` values. It is better to do it programmatically by transforming `ArrowSerializationOptions.CompressionCodec.values()`. In our case though, this field can be deleted (see below)

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java [hunk 3]
```diff
@@ -294,6 +306,19 @@ public static SparkBigQueryConfig from(
             .transform(Integer::parseInt)
             .or(MIN_STREAMS_PER_PARTITION);
 
+    String arrowCompressionCodecParam =
+        getAnyOption(globalOptions, options, ARROW_COMPRESSION_CODEC_OPTION)
+            .transform(String::toUpperCase)
+            .or(DEFAULT_ARROW_COMPRESSION_CODEC.toString());
+
+    if (!PERMITTED_ARROW_COMPRESSION_CODECS.contains(arrowCompressionCodecParam)) {
```
Comment (davidrabinowitz) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java:314]: Can be replaced with `ArrowSerializationOptions.CompressionCodec.valueOF()`

File: spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/SparkBigQueryConfigTest.java [hunk 4]
```diff
@@ -166,6 +170,37 @@ public void testConfigFromOptions() {
...
+    DataSourceOptions options =
+        new DataSourceOptions(
+            ImmutableMap.<String, String>builder()
+                .put("table", "test_t")
+                .put("dataset", "test_d")
+                .put("project", "test_p")
+                .put("arrowCompressionCodec", "randomCompression")
+                .build());
+
+    try {
```
Comment (davidrabinowitz) [inline spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/SparkBigQueryConfigTest.java:188]: Better to use `Assert.assertThrows()`, see [here for example](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/c2eeaf096c28e25af1929276c1d387d12c10fd3b/spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/SparkBigQueryProxyAndHttpConfigTest.java#L265)

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (VinayLondhe14) [reply README.md]: Removed this. WIll open a new PR after release

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java (hunk 2 above)
Comment (VinayLondhe14) [reply spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java]: Good call. Removed this.

**Review by emkornfield — APPROVED** [review]

**Review by davidrabinowitz — APPROVED** [review]
